### ISSUE TITLE
[Core feature] Investigate how to turn the PyTorch visualizer into a deck

### ISSUE THREAD

**Issue by eapolinario** [flyteorg/flyte#6123]
### Motivation: Why do you think this is important?

Huggingface announced a pytorch visualizer: https://huggingface.co/blog/train_memory. Now that we're approaching the ability to generate decks for failed executions, we should investigate if possibility of turning the pytorch visualizer into a deck.

### Goal: What should the final outcome look like, ideally?

Investigate the possibility of rendering the profiling data generated by pytorch in a flyte deck.

### Describe alternatives you've considered 

Copy the profiling file to blob store and generate a link at the end of the execution. 

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

**Comment by 10sharmashivam** [comment]
Hi @eapolinario @davidmirror-ops, I’d like to pick this issue.

**Comment by davidmirror-ops** [comment]
Thank you Shivam!
@eapolinario is this dependent on streaming decks?

**Comment by eapolinario** [comment]
> @eapolinario is this dependent on streaming decks?
No, but it'll make a lot more sense to have this deck in failed executions. We shouldn't need anything specific to that though.

**Comment by 10sharmashivam** [comment]
Hi @eapolinario @davidmirror-ops, I have submitted [PR](https://github.com/flyteorg/flytekit/pull/3105) for this issue. Kindly have a look, and let me know of any changes or feedback.

Regards
